duckduckgo/Android
# Update Anvil and contribute multibindings for workers

Task/Issue URL: https://app.asana.com/0/414730916066338/1200105747936521/f
Tech Design URL: 
CC: 

**Description**:
This PR updates anvil to version 2.2.0. Anvil now provides a new handy annotation `ContributesMultibinding` that allow us to contribute dependencies into sets, like we do for eg. workers or view model factories. 
Using that annotation allow us to remove a bunch of boilerplate code.

In this PR we:
- Bumped anvil to 2.2.0 and upgrade dagger to 2.32
- Get VM factories to contribute multibinding
- Get Worker factories to contribute multibindings

**Steps to test this PR**:
1. app builds
2. install/upgrade app
3. verify bookmarks work
4. verify fireproof sites work
5. verify search/navigation works
6. verify fire button works



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
